File: 프로그래머스/2/42839. 소수 찾기/소수 찾기.py
from itertools import permutations

def solution(numbers):
    answer = 0
    nums = [int(c) for c in numbers]
    p = permutations(nums)
    for c in range(1, len(nums)+1):
        p = list(set(permutations(nums, c)))
        for k in p:
            if k[0] == 0:
                continue
            n = int(''.join(str(x) for x in k))
            
            if isPrime(n):
                answer += 1
    return answer

def isPrime(num):
    i = 2
    if num < 2:
        return False
    elif num == 2:
        return True
    while i < num:
        if num % i == 0:
            return False
        i += 1
    
    return True

File: 프로그래머스/1/12921. 소수 찾기/소수 찾기.java
class Solution {
    public int solution(int n) {
        int answer = 0;
        for (int k = 1; k <= n; k++) {
            boolean isDecimal = true;
             if (k == 1) {
                    continue;
            }
            for(int i = 2; i * i <= k; i++) {
                if (k % i == 0) {
                    isDecimal = false;
                    break;
                }
             }
            if (isDecimal) {
                answer++;
            }
        }
        return answer;
    }
}